articles/libraries/auth0js/v9/migration-guide.md

```diff
@@ -0,0 +1,55 @@
+---
+section: libraries
+title: Migrating to Auth0.js v9
+description: How to migrate to Auth0.js v9
+toc: true
+---
+# Migrating to Auth0.js v9
+
+Auth0.js v9 has been improved to work better in embedded login scenarios. It operates with enhanced security and removes dependencies that have been deprecated as per Auth0's roadmap. In some cases, these security enhancements may impact application behavior when upgrading from an earlier versions of Auth0.js. 
+
+Auth0.js can be used to implement authentication in different ways:
+
+- Centralized Login, using the `.authorize()` method, where the user is redirected from your website to an Auth0's Hosted Login Page.
+
+- Embedded Login, using the `.login()`, `popup.loginWithCredentials()`, `client.loginWithCredentials()` methods, where the login dialog is displayed in the application's website.
+
+- In the [Hosted Login Page](/hosted-pages/login), where you can use the same methods as in Embedded Login but from inside a customized Auth0's Hosted Login Page. Most customers don't customize Auth0 Hosted Login Page with Auth0.js, so your probably don't need to worry about this scenario.
+
+Depending on how you are using Auth0.js, you have different options:
+
+| Scenario | Migration to v9 | 
+| --- | --- | 
+| Centralized Login | Recommended |
+| Embedded Login | Required |
+| Customized Hosted Page | Not Supported |
+
```

Comment:
Again, I would replace with:

| In your app, to trigger Centralized Login | Recommended |
| In your app, to implement Embedded Login | Required |
| In a customized Hosted Login Page | Not Supported, keep using Auth0.js v8 and/or Lock v10 |